project-management-infrastructure/src/main/java/life/qbic/projectmanagement/infrastructure/dataset/LocalRawDatasetRepositoryImpl.java

```diff
@@ -111,6 +123,177 @@ public List<RawDatasetInformationNgs> findAllNgs(String experimentId, int offset
+    fullSpec = fullSpec.and(sampleNameSpec).and(experimentIdSpec);
+
+
+    return ngsInfoRepository.findAll(fullSpec,
+            new OffsetBasedRequest(offset, limit, Sort.by(orders))).getContent()
+        .stream().map(LocalRawDatasetRepositoryImpl::convert).toList();
+  }
+
+  private static Order fromAPItoJpaRawData(
+      AsyncProjectService.SortOrder<RawDataSortingKey> sortOrder) {
+    var mappedJpaProperty = propertyFromAPItoJpaRawDataset(sortOrder.key()).orElseThrow(
+        () -> new IllegalArgumentException(
+            "Cannot map key to JPA raw dataset property: " + sortOrder.key()
+        ));
+    if (sortOrder.direction() == SortDirection.ASC) {
+      return Order.asc(mappedJpaProperty);
+    } else {
+      return Order.desc(mappedJpaProperty);
+    }
+  }
+
+  private static Optional<String> propertyFromAPItoJpaRawDataset(RawDataSortingKey key) {
+    return Optional.ofNullable(rawDataSortingMap.getOrDefault(key, null));
+  }
+
+  @Override
+  public List<RawDatasetInformationPxP> findAllPxP(String experimentId, int offset, int limit,
+      RawDatasetFilter filter) throws LookupException {
+    List<Order> orders;
+
+    try {
+      orders = filter.sortOrders().stream().map(LocalRawDatasetRepositoryImpl::fromAPItoJpaRawData)
+          .toList();
+    } catch (IllegalArgumentException e) {
+      throw new LookupException("Lookup for raw datasets failed", e);
+    }
+
+    Specification<LocalRawDatasetPxpEntry> sampleNameSpec = (root, query, builder) -> {
+      if (filter.filterTerm().isBlank()) {
+        return builder.conjunction();
+      }
+      Expression<String> function = builder.function("JSON_SEARCH", String.class, root.get("measuredSamples"),
+          builder.literal("one"),
+          builder.literal("%" + filter.filterTerm() + "%"),
+          builder.nullLiteral(String.class),
+          builder.literal("$[*].label")
+          );
+
+      return builder.isNotNull(function);
+    };
+
+    Specification<LocalRawDatasetPxpEntry> experimentIdSpec = (root, query, builder) -> {
+      if  (experimentId == null) {
+        return builder.conjunction();
+      }
+      return builder.equal(root.get("experimentId").as(String.class), experimentId);
+    };
+
+    Specification<LocalRawDatasetPxpEntry> fullSpec = Specification.unrestricted();
+    fullSpec = fullSpec.and(sampleNameSpec).and(experimentIdSpec);
+
+    return pxpInfoRepository.findAll(fullSpec,
+            new OffsetBasedRequest(offset, limit, Sort.by(orders))).getContent()
+        .stream().map(LocalRawDatasetRepositoryImpl::convert).toList();
+  }
+
+  @Override
+  public Integer countNGS(String experimentId, RawDatasetFilter filter) throws LookupException {
+
+    Specification<LocalRawDatasetNgsEntry> sampleNameSpec = (root, query, builder) -> {
+      if (filter == null) {
+        return builder.conjunction();
+      }
+      Expression<String> function = builder.function("JSON_SEARCH", String.class, root.get("measuredSamples"),
```

Comment:
The mapping of the filter to the json search repeats a lot. As this function is not strongly linked to the domain or intention (does not matter if you want to retrieve or count, does not matter if pxp or ngs) maybe this shared functionality could be extracted into some method?